src/h/uCLib

```diff
@@ -0,0 +1,81 @@
+ *           C Library to use on RISC OS to build utils (transient)
+ *           executables  using  the C  Programming  language  and,
+ *           hopefully, C++ as well.
+ *           Please note: This library is NOT a replacement for the
+ *           Acorn/ROOL  SharedCLib (SCL)  or  the  ANSILib (ANSL).
+ *           It is  intended for use  with applications that do not
+ *           need the  entire  SCL or ANSL,  that  wish to  have no
+ *           dependecies  on  the  SCL/ANSL and still use C for the
+ *           source coding.
+ *
+ *   author: Paolo Fabio Zaino
+ * Copyright 2021 by Paolo Fabio Zaino
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions, the distribution license and the
+ * following disclaimer.
+ * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials
+ * provided with the distribution.
+ * Neither the name of the author nor the names of the contributors
+ * may be used to endorse or promote products derived from this
+ * software without specific prior written permission.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License
+ * the restrictions above and the disclaimer below.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ * OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ */
+
+#define  UNUSED(x)              (x = x)
+
+#define OS_WriteC    0x00
+#define OS_WriteS    0x01
+#define XOS_WriteS   0x20001
+#define OS_Write0    0x02
+#define XOS_Write0   0x20002
+#define OS_NewLine   0x03
+#define XOS_NewLine  0x20003
+#define OS_Exit      0x11
+#define XOS_Exit     0x20011
+
+#define ABEX         0x58454241
+
+void sys_exit(int error_code);
+
+int sys_write_stdout(const char *buf, unsigned count);
+
+void sys_write_newline(void);
```

Comment:
These two functions, as they stand, are already in OSLib, unless you are planning on doing \n -> \n\r expansion. Better to write a putchar and then a printf if that's desired?

Reply:
Yes, when I have a good entry/s I'll start implementing the C functions in uCLib correctly and yes there will be a putchar etc..., for now sys_write serves the purpose of testing what's happening in the code.